### ISSUE TITLE
wat2wasm: float nan hex literals with underscores are reported as invalid

### ISSUE THREAD

**Issue by errikos** [WebAssembly/wabt#1104]
Example test case from the spec test suite ([const.wast#L147](https://github.com/WebAssembly/testsuite/blob/master/const.wast#L147)):
```wasm
(module
    (func
        (f32.const nan:0x7f_ffff)
        drop
    )
)
```

According to the [specification](https://webassembly.github.io/spec/core/text/values.html#floating-point), the above program seems perfectly valid, however the `wat2wasm` tool reports:

```
a.wat:3:20: error: invalid literal "nan:0x7f_ffff"
        (f32.const nan:0x7f_ffff)
                   ^^^^^^^^^^^^^
```
If the underscore is removed, then compilation is successful.

The same problem exists in the `wat2wasm` [online demo](https://webassembly.github.io/wabt/demo/wat2wasm/) as well.

From what I understand, the top-level module S-expressions like the above (i.e. the ones which are not a part of an `assert_` S-expression), are supposed to be valid. How does this pass the spec test suite?

Cheers,
Ergys

**Comment by binji** [comment]
Yes, these are recently modified tests that wabt doesn't parse properly. I'm in the process of fixing, but haven't finished yet.

**Comment by binji** [comment]
This was fixed in #1111.
